sourcegraph/sourcegraph
# Syntactic indexing: add database schema and dbworker

Closes #59794 

This PR:

- Adds a basic schema for syntactic code intel indexes (up and down migration)
- Sets up everything required to use `dbworker` interface
- Adds a stub indexing worker implementation
- Adds tests to ensure conformance with dbworker interface (run against real postgres)

## Test plan

New tests to ensure schema conformance.

For manual testing:
1. Start the configuration: `sg start codeintel-syntactic`
2. Execute this SQL query: `insert into syntactic_scip_indexes(id, commit, repository_id, outfile) values (nextval('syntactic_scip_index_id_seq'), '8c0cc755be302107b675824b0fb1b985992092b8', 1, 'index.scip');`
3. Observe the worker pick up the job:

```
[syntactic-c...1] INFO syntactic-code-intel-worker workerutil/worker.go:329 Dequeued record for processing {"TraceId": "a0b9c4a988d9e46ac42b8c4deb63d670", "SpanId": "6483b7e47fa617f2", "name": "syntactic_code_intel_indexer", "id": "1"}
[syntactic-c...1] INFO syntactic-code-intel-worker.Handle shared/indexing_worker.go:34 Stub indexing worker handling record {"TraceId": "a0b9c4a988d9e46ac42b8c4deb63d670", "SpanId": "470708143297b7ba", "handle": {"id": 1, "repository name": "github.com/sourcegraph-testing/zap", "commit": "8c0cc755be302107b675824b0fb1b985992092b8"}}
```
